rfcs/0000-poa-testnet/0000-poa-testnet.md

```diff
@@ -0,0 +1,183 @@
+
+The protocol uses `n % VALIDATOR_COUNT` as the index to choose the attester from validators; each validator checks the index and decides whether to attest a new block itself or wait for another attester to produce a new block.
+
+However, an in-turned attester may fail to produce a block due to network error or other reasons; in this case, other validators must produce the new block to make the POA testnet continue.
+
+Validators can use a simple strategy:
+
+1. For block height `n` a validator checks `n % VALIDAROR_COUNT`.
+2. If `INDEX == n % VALIDATOR_COUNT`, which means the validator is in it's turn to attests block `n`. The validator should wait for `BLOCK_INTERVAL` seconds then attests the block `n` with difficulty set to `2`.
+3. If `INDEX != n % VALIDATOR_COUNT`, which means the validator is not in it's turn to attests block `n`. The validator should wait for `BLOCK_INTERVAL + rand(VALIDATOR_COUNT) * 0.5` seconds to wait for another attester to produce a new block. If there are no new block produced during the time, the validator should attest a new block with difficulty set to `1`.
+4. If the validator is the attester of the last block, wait for `ATTEST_INTERVAL` blocks then continue this strategy.
+
+Notice the difficulty set to `2` when an in-turn attester produces a block and set to `1` when a not in-turn attester produces a block. This makes the in-turn attester's block total difficulty higher than the not in-turn attester's block; once two validators attest at the same height due to the network error, the other nodes still chooses the higher total difficulty block as the main chain.
+
+`ATTEST_INTERVAL` is used to preventing malicious validators censor the POA network. A validator cannot attest two blocks within the `ATTEST_INTERVAL` number.  Consider the situation that malicious validators try to control the whole network: 
+
+The first malicious validator must wait for `ATTEST_INTERVAL + 1` blocks to produce a block again. When the number of malicious validators is less than `ATTEST_INTERVAL + 1`, there must exist an honest validator who has the chance to submit a block includes majority voting to evict malicious validators.
+
+`ATTEST_INTERVAL` can be set to `VALIDATOR_COUNT / 2` the honest validators could eventually evict malicious validators unless the half of validators corrupted.
+
+### validator list and on-chain governance
+
+As previously described, the POA testnet maintained by a group of validators. We have a build-in on-chain governance mechanism to allows updating the exists validator list. The on-chain governance mechanism designed to be loose; it allows several validators that greater than `VOTE_LIMIT` to sign a transaction together that updating the exists validators to a list of new validators.
```

Comment:
```suggestion
As previously described, the POA testnet maintained by a group of validators. We have a build-in on-chain governance mechanism to update the exists validator list. The on-chain governance mechanism designed to be loose; it allows several validators that greater than `VOTE_LIMIT` to sign a transaction together that updating the exists validators to a list of new validators.
```

Reply:
What does 'designed to be loose` mean here?

Reply:
It means that the on-chain governance contract has very few verification rules, I think I can just remove this.